### PR TITLE
Whitelist text formatting tags in :html responses

### LINKED ISSUE #88: "km2" misindicates square kilometers
Here's what WikiBase currently returns:

    $ telnet nauru 8023
    (get "wikipedia-sea" "Black Sea" (:code "AREA"))
    ((:html "436,402&#160;km2 (168,500&#160;sq&#160;mi)"))

Here's the wikimarkup:

    | area               = {{convert|436402|km2|sqmi|-2|abbr=on}}

Here's the HTML source:

    <td>436,402&#160;km<sup>2</sup> (168,500&#160;sq&#160;mi)</td>

Shouldn't WikiBase return this?:

    $ telnet nauru 8023
    (get "wikipedia-sea" "Black Sea" (:code "AREA"))
    ((:html "436,402&#160;km<sup>2</sup> (168,500&#160;sq&#160;mi)"))

(Or alternatively `...436,402 sq km` or `...436,402 square kilometers`, but that seems like unhelpful complication of code.)

### PR DESCRIPTION
**Pull request by fakedrake**
Infoboxes clean up the html of the values removing all tags but a few. Added some text formatting tags to the whitelist.

Fixes #88

### CODE REVIEW

**Comment by alvaromorales** [comment]
Looks good! Interesting fix.

> The reasoning behind the code there is that we remove all markup to make sure no weird stuff like this sneaks into the answer and whitelist only tags that we know are good, specifically li, ul and br

Do you have any examples of bad markup that might sneak it? I think we need a code comment that explains _why_ we're stripping markup, not just how. It should also explain why we whitelist some tags.

> If we go with the whitelist approach, other text formatting tags that probably should be whitelisted are `<b> <em> <i> <small> <strong> <sub> <sup> <ins> <del> <mark>`

This list looks pretty exhaustive to me, but I haven't checked the HTML spec. Do you think it's complete?

This change needs more tests. Have you tried `(get "wikipedia-president" "Bill Clinton" "RELIGION")`? Does the `[1]` reference get included?

**Comment by alvaromorales** [comment]
The Beethoven and Mozart test cases were already passing previous to this change.

This PR proposes to pick the date / date range "with the least missing values". What does that mean exactly? Please give some examples of Wikipedia articles to understand how this improves precision.

I don't think this PR solves the underlying issue with dates in WikipediaBase, as described by @sfelshin in #94.

> It looks like the first-sentence date parsing favors false positives over false negatives, and it should be the other way around, because the Infolab prioritizes precision over recall.

WikipediaBase's current approach is to _always_ return :yyyymmdd for dates. START only uses the :yyymmdd answer for numeric calculations, e.g. "Was Mozart born before Beethoven?" For non-numeric calculations START is happy to show an :html answer.

For dates, instead of just returning:

```
(get "wikibase-person" "Ludwig van Beethoven" (:CODE "BIRTH-DATE"))
((:yyyymmdd 17701217))
```

we should return

```
(get "wikibase-person" "Ludwig van Beethoven" (:CODE "BIRTH-DATE"))
((:html "baptised 17 December 1770" :yyyymmdd 17701217))
```

For cases that are too hard to parse, we should avoid returning :yyyymmdd and only return :html. Is it possible to return :yyyymmdd only if we're absolutely confident?

**Comment by fakedrake** [comment]
I agree about the tests. I committed some documentation about the tags. 

We should probably remove all annotations from infobox values as they are contextual and make a backlog issue to find a better way to handle them.

I got the element list from [w3schools](http://www.w3schools.com/html/html_formatting.asp). Since the site is educational and not comprehensive, and the html4 specification has information about text formatting [a](http://www.w3.org/TR/html401/present/graphics.html#h-15.2) [bit](http://www.w3.org/TR/html401/struct/text.html#h-9.2.1) spread I didn't provide a link in the comments.

**Comment by alvaromorales** [comment]
I recently removed files stored on Git LFS from the repository. I'm going to have to close this PR because your fork references the old git history, and merging would dirty the history. I apologize for the inconvenience.

This change is almost almost there; I'll ask one of our students to add tests so we can merge this in.

**Comment by alvaromorales** [comment]
I apologize -- I just noticed the latest commit with unit tests. This change is ready to be merged and looks good.

@cordeliaavery recently made changes to the HTML parsing code, which explains the merge conflicts. I'll ask her to de-conflict this patch and merge it in.
